fs_permissions_diff: Strip commit hash from file name

pratheekshasn opened the pull request:
### Summary of Changes

While comparing files across different builds, the version commit hashes in the file names are not currently stripped off before comparing.

This PR adds the extra regex that matches the commit hashes in the file names.


### Justification

[AB#3014543](https://dev.azure.com/ni/DevCentral/_workitems/edit/3014543)


### Testing

* [x] I have built the core package feed with this PR in place. (`bitbake packagefeed-ni-core`)
* [x] I have built and installed the new changes and tested on a target, and the logs added because of the difference in file names caused by the commit hash are no longer present.
* [x] The test has been run on the current kernel and the next kernel.



### Procedure

* [x] I certify that the contents of this pull request complies with the [Developer Certificate of Origin](https://github.com/ni/nilrt/blob/HEAD/docs/CONTRIBUTING.md#developer-certificate-of-origin-dco).

chaitu236 commented on recipes-ni/ni-base-system-image-tests/files/fs_permissions_diff.py:
```diff
@@ -207,7 +207,7 @@ def strip_versions_from_path(path):
     stripped_path = re.sub(r"salt-\d+\.\d+\+\d+\.g[0-9a-f]+", "salt-SALTVERSION", stripped_path)
     # Strip everything that looks like an RT kernel version (so module and kernel version changes are ignored)
-    stripped_path = re.sub(r"\d+\.\d+\.\d+-rt\d+", "VERSION.VERSION.VERSION-rtVERSION", stripped_path)
+    stripped_path = re.sub(r"\d+\.\d+\.\d+-rt\d+-?(next)?-?g?[0-9a-f]*", "VERSION.VERSION.VERSION-rtVERSION", stripped_path)
```
Isn't following slightly better?

r"\d+\.\d+\.\d+-rt\d+(-next)?(-g[a-f0-9]+)?" [source: copilot]

pratheekshasn replied:
I have fixed this now.

chaitu236 reviewed (commented):
Did you test both current and -next kernels?

chaitu236 commented on recipes-ni/ni-base-system-image-tests/files/fs_permissions_diff.py:
```diff
@@ -207,7 +207,7 @@ def strip_versions_from_path(path):
     stripped_path = re.sub(r"salt-\d+\.\d+\+\d+\.g[0-9a-f]+", "salt-SALTVERSION", stripped_path)
     # Strip everything that looks like an RT kernel version (so module and kernel version changes are ignored)
-    stripped_path = re.sub(r"\d+\.\d+\.\d+-rt\d+", "VERSION.VERSION.VERSION-rtVERSION", stripped_path)
+    stripped_path = re.sub(r"\d+\.\d+\.\d+-rt\d+(-next)?(-g[a-f0-9]+)?", "VERSION.VERSION.VERSION-rtVERSION", stripped_path)
```
I think it's better to have two regexes - one for current kernel and one for -next kernel; and not strip the "-next" string in the next kernel one. This way there won't be collisions in stripped paths when the -next kernel is same version as current kernel.

pratheekshasn replied:
I have changed this now, and have tested it on both the current kernel and the next kernel.

chaitu236 replied:
Since we have separate lines for -next and current, the next's regex should not have `(-next)` as optional and replaced string should also have -next. i.e., something like (not tested by me)

`re.sub(r"\d+\.\d+\.\d+-rt\d+(-next)(-g[a-f0-9]+)?", "VERSION.VERSION.VERSION-rtVERSION-next", stripped_path)`

pratheekshasn replied:
When we compare the current kernel files against the next kernel, this would show up as a diff where the only difference would be the addition of the word "-next", it would just cause noise in the test log.

chaitu236 replied:
I hadn't realized that we can sometimes compare -next kernels with current kernels. Thanks for catching that

CommitSentience committed:
fs_permissions_diff: Strip commit hash from file name

While comparing files across different builds, the version commit hashes
in the file names are not currently stripped off before comparing.

This PR adds the extra regex that matches the commit hashes in the
file names.

Signed-off-by: Pratheeksha S N <pratheeksha.s.n@emerson.com>
